nat/service_ics.go

```diff
@@ -206,6 +219,57 @@ func (nat *serviceICS) getInternalInterfaceName() (string, error) {
 	return ifaceName, nil
 }
 
+func (nat *serviceICS) interfaceIPWatchdog() {
+	for {
+		select {
+		case <-time.After(5 * time.Second):
+			for ifaceName, rule := range nat.ifaces {
+				_, ipnet, err := net.ParseCIDR(rule.SourceAddress)
+				if err != nil {
+					log.Warnf("%s Failed to parse IP-address: %s", natLogPrefix, rule.SourceAddress)
+					continue
+				}
+
+				incrementIP(ipnet.IP)
+				if err := nat.setInterfaceAddr(ifaceName, ipnet); err != nil {
+					log.Warnf("%s Failed to assign IP-address '%s' to interface '%d'", natLogPrefix, ipnet.String(), ifaceName)
+				}
+			}
+		case <-nat.ipAddrWatchdog:
+			return
+		}
+	}
+}
+
+func (nat *serviceICS) setInterfaceAddr(name string, subnet *net.IPNet) error {
```

Comment:
`nat` is very wrong receiver name for `service..`:
- it's coupled with interface it's implementing, but interfaces should be on caller side only
- it tends to conflict with some package

Reply:
renamed